File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-com-5-4/PercentageTolerance.cpp
// PercentageTolerance.cpp 
#include "PercentageTolerance.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

PercentageTolerance::PercentageTolerance(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //postitiveNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* postitiveNode = xmlNode->FirstChildElement("postitive");

   if(postitiveNode){postitiveIsNull_ = false;}
   else{postitiveIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- postitiveNode , address : " << postitiveNode << std::endl;
   #endif
   if(postitiveNode)
   {
      if(postitiveNode->Attribute("href") || postitiveNode->Attribute("id"))
      {
          if(postitiveNode->Attribute("id"))
          {
             postitiveIDRef_ = postitiveNode->Attribute("id");
             postitive_ = boost::shared_ptr<RestrictedPercentage>(new RestrictedPercentage(postitiveNode));
             postitive_->setID(postitiveIDRef_);
             IDManager::instance().setID(postitiveIDRef_,postitive_);
          }
          else if(postitiveNode->Attribute("href")) { postitiveIDRef_ = postitiveNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { postitive_ = boost::shared_ptr<RestrictedPercentage>(new RestrictedPercentage(postitiveNode));}
   }

   //negativeNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* negativeNode = xmlNode->FirstChildElement("negative");

   if(negativeNode){negativeIsNull_ = false;}
   else{negativeIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- negativeNode , address : " << negativeNode << std::endl;
   #endif
   if(negativeNode)
   {
      if(negativeNode->Attribute("href") || negativeNode->Attribute("id"))
      {
          if(negativeNode->Attribute("id"))
          {
             negativeIDRef_ = negativeNode->Attribute("id");
             negative_ = boost::shared_ptr<RestrictedPercentage>(new RestrictedPercentage(negativeNode));
             negative_->setID(negativeIDRef_);
             IDManager::instance().setID(negativeIDRef_,negative_);
          }
          else if(negativeNode->Attribute("href")) { negativeIDRef_ = negativeNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { negative_ = boost::shared_ptr<RestrictedPercentage>(new RestrictedPercentage(negativeNode));}
   }

   //optionNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* optionNode = xmlNode->FirstChildElement("option");

   if(optionNode){optionIsNull_ = false;}
   else{optionIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- optionNode , address : " << optionNode << std::endl;
   #endif
   if(optionNode)
   {
      if(optionNode->Attribute("href") || optionNode->Attribute("id"))
      {
          if(optionNode->Attribute("id"))
          {
             optionIDRef_ = optionNode->Attribute("id");
             option_ = boost::shared_ptr<PartyReference>(new PartyReference(optionNode));
             option_->setID(optionIDRef_);
             IDManager::instance().setID(optionIDRef_,option_);
          }
          else if(optionNode->Attribute("href")) { optionIDRef_ = optionNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { option_ = boost::shared_ptr<PartyReference>(new PartyReference(optionNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<RestrictedPercentage> PercentageTolerance::getPostitive()
{
   if(!this->postitiveIsNull_){
        if(postitiveIDRef_ != ""){
             return boost::shared_static_cast<RestrictedPercentage>(IDManager::instance().getID(postitiveIDRef_));
        }else{
             return this->postitive_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<RestrictedPercentage>();
   }
}
boost::shared_ptr<RestrictedPercentage> PercentageTolerance::getNegative()
{
   if(!this->negativeIsNull_){
        if(negativeIDRef_ != ""){
             return boost::shared_static_cast<RestrictedPercentage>(IDManager::instance().getID(negativeIDRef_));
        }else{
             return this->negative_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<RestrictedPercentage>();
   }
}
boost::shared_ptr<PartyReference> PercentageTolerance::getOption()
{
   if(!this->optionIsNull_){
        if(optionIDRef_ != ""){
             return boost::shared_static_cast<PartyReference>(IDManager::instance().getID(optionIDRef_));
        }else{
             return this->option_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<PartyReference>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-com-5-4/PercentageTolerance.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class PercentageTolerance : ISerialized
    {
        public PercentageTolerance(XmlNode xmlNode)
        {
            XmlNode postitiveNode = xmlNode.SelectSingleNode("postitive");
            
            if (postitiveNode != null)
            {
                if (postitiveNode.Attributes["href"] != null || postitiveNode.Attributes["id"] != null) 
                {
                    if (postitiveNode.Attributes["id"] != null) 
                    {
                        postitiveIDRef_ = postitiveNode.Attributes["id"].Value;
                        RestrictedPercentage ob = new RestrictedPercentage(postitiveNode);
                        IDManager.SetID(postitiveIDRef_, ob);
                    }
                    else if (postitiveNode.Attributes["href"] != null)
                    {
                        postitiveIDRef_ = postitiveNode.Attributes["href"].Value;
                    }
                    else
                    {
                        postitive_ = new RestrictedPercentage(postitiveNode);
                    }
                }
                else
                {
                    postitive_ = new RestrictedPercentage(postitiveNode);
                }
            }
            
        
            XmlNode negativeNode = xmlNode.SelectSingleNode("negative");
            
            if (negativeNode != null)
            {
                if (negativeNode.Attributes["href"] != null || negativeNode.Attributes["id"] != null) 
                {
                    if (negativeNode.Attributes["id"] != null) 
                    {
                        negativeIDRef_ = negativeNode.Attributes["id"].Value;
                        RestrictedPercentage ob = new RestrictedPercentage(negativeNode);
                        IDManager.SetID(negativeIDRef_, ob);
                    }
                    else if (negativeNode.Attributes["href"] != null)
                    {
                        negativeIDRef_ = negativeNode.Attributes["href"].Value;
                    }
                    else
                    {
                        negative_ = new RestrictedPercentage(negativeNode);
                    }
                }
                else
                {
                    negative_ = new RestrictedPercentage(negativeNode);
                }
            }
            
        
            XmlNode optionNode = xmlNode.SelectSingleNode("option");
            
            if (optionNode != null)
            {
                if (optionNode.Attributes["href"] != null || optionNode.Attributes["id"] != null) 
                {
                    if (optionNode.Attributes["id"] != null) 
                    {
                        optionIDRef_ = optionNode.Attributes["id"].Value;
                        PartyReference ob = new PartyReference(optionNode);
                        IDManager.SetID(optionIDRef_, ob);
                    }
                    else if (optionNode.Attributes["href"] != null)
                    {
                        optionIDRef_ = optionNode.Attributes["href"].Value;
                    }
                    else
                    {
                        option_ = new PartyReference(optionNode);
                    }
                }
                else
                {
                    option_ = new PartyReference(optionNode);
                }
            }
            
        
        }
        
    
        #region Postitive_
        private RestrictedPercentage postitive_;
        public RestrictedPercentage Postitive_
        {
            get
            {
                if (this.postitive_ != null)
                {
                    return this.postitive_; 
                }
                else if (this.postitiveIDRef_ != null)
                {
                    postitive_ = IDManager.getID(postitiveIDRef_) as RestrictedPercentage;
                    return this.postitive_; 
                }
                else
                {
                      return this.postitive_; 
                }
            }
            set
            {
                if (this.postitive_ != value)
                {
                    this.postitive_ = value;
                }
            }
        }
        #endregion
        
        public string postitiveIDRef_ { get; set; }
        #region Negative_
        private RestrictedPercentage negative_;
        public RestrictedPercentage Negative_
        {
            get
            {
                if (this.negative_ != null)
                {
                    return this.negative_; 
                }
                else if (this.negativeIDRef_ != null)
                {
                    negative_ = IDManager.getID(negativeIDRef_) as RestrictedPercentage;
                    return this.negative_; 
                }
                else
                {
                      return this.negative_; 
                }
            }
            set
            {
                if (this.negative_ != value)
                {
                    this.negative_ = value;
                }
            }
        }
        #endregion
        
        public string negativeIDRef_ { get; set; }
        #region Option_
        private PartyReference option_;
        public PartyReference Option_
        {
            get
            {
                if (this.option_ != null)
                {
                    return this.option_; 
                }
                else if (this.optionIDRef_ != null)
                {
                    option_ = IDManager.getID(optionIDRef_) as PartyReference;
                    return this.option_; 
                }
                else
                {
                      return this.option_; 
                }
            }
            set
            {
                if (this.option_ != value)
                {
                    this.option_ = value;
                }
            }
        }
        #endregion
        
        public string optionIDRef_ { get; set; }
        
    
        
    
    }
    
}

